### PR TITLE
Fix s390x rax2

### PR DESCRIPTION
**Pull request by eagleoflqj**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

A step forward to solve #16168.

**Test plan**

https://travis-ci.com/github/eagleoflqj/radare2/jobs/442084042 is not readable, but on my qemu it was 10XXs, now is
```
liumeo@liumeo-s390x:~/radare2/test$ r2r db/tools/rax2 
Running from /home/liumeo/radare2/test
Loaded 67 tests.
Skipping json tests because jq is not available.
[67/67]                    66 OK         1 BR        0 XX        0 FX
Finished in 26 seconds.
```

**Closing issues**

...

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
It broke other systems though:
```
 [XX] db/tools/r2 no preincrement/predecrement in 3rd part of for statement
R2_NOPLUGINS=1 radare2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc '"!git grep \"for[[:space:]]*([^;)]*;[^;)]*;\([^,)(]*,\)*[[:space:]]*\(++\|--\)[[:space:]]*[[:alpha:]_][[:alnum:]_]*[[:space:]]*[,)]\" -- .. :!db/tools/r2 :!../shlr/** :!../libr/asm/arch/** :!../**/mangling/cxx/cp-demangle.c"' -
-- stdout
@@ -0,0 +1,4 @@
+../libr/main/rax2.c:				for (int i = 0; i < 8; ++i) {
+../libr/main/rax2.c:				for (int i = 0; i < 4; ++i) {
+../libr/main/rax2.c:				for (int i = 0; i < 8; ++i) {
+../libr/main/rax2.c:				for (int i = 0; i < 4; ++i) {
```
https://github.com/radareorg/radare2/pull/17934/checks?check_run_id=1416507953#step:13:495

File: libr/main/rax2.c [hunk 1]
```diff
@@ -55,7 +55,10 @@ static int format_output(RNum *num, char mode, const char *s, int force_mode, ut
 	}
 	if (flags & 2) {
 		ut64 n2 = n;
-		r_mem_swapendian ((ut8 *) &n, (ut8 *) &n2, (n >> 32)? 8: 4);
+		r_mem_swapendian ((ut8 *) &n, (ut8 *) &n2, 8);
+		if (!(int) n) {
```
Comment (trufae) [inline libr/main/rax2.c:59]: Why the cast

File: libr/main/rax2.c [hunk 2]
```diff
@@ -311,24 +315,26 @@ static int rax(RNum *num, char *str, int len, int last, ut64 *_flags, int *fm) {
...
 			/* is 64 bit value */
-			ut8 *np = (ut8 *) &n;
 			if (flags & 1) {
 				fwrite (&n, sizeof (n), 1, stdout);
 			} else {
-				printf ("%02x%02x%02x%02x"
-					"%02x%02x%02x%02x\n",
-					np[0], np[1], np[2], np[3],
-					np[4], np[5], np[6], np[7]);
+				for (int i = 0; i < 8; ++i) {
```
Comment (trufae) [inline libr/main/rax2.c:321]: Dont use for(int. Use int;for. Dont define variables inside the for statment

Comment (trufae) [reply libr/main/rax2.c:321]: Also never use preincrement. Unless its the only way to do it. In this place is unnecessary

File: libr/main/rax2.c [hunk 3]
```diff
@@ -311,24 +315,26 @@ static int rax(RNum *num, char *str, int len, int last, ut64 *_flags, int *fm) {
...
+					printf ("%02x", (int) n & 0xff);
```
Comment (trufae) [inline libr/main/rax2.c:322]: I would add () to cast the whole thing. Ot just the n

File: libr/main/rax2.c (hunk 1 above)
Comment (eagleoflqj) [reply libr/main/rax2.c:59]: If n was 32 bit, after swapping endian its lower 32 bits are zero.

**Review by XVilka — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**